src/Features/CSharp/Portable/ChangeSignature/CSharpChangeSignatureService.cs

```diff
@@ -29,9 +29,9 @@ internal sealed class CSharpChangeSignatureService : AbstractChangeSignatureServ
             var tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
             var token = tree.GetRoot(cancellationToken).FindToken(position != tree.Length ? position : Math.Max(0, position - 1));
 
-            var ancestorDeclarationKinds = restrictToDeclarations ? _invokableAncestorKinds.Add(SyntaxKind.Block) : _invokableAncestorKinds;
+            var ancestorDeclarationKinds = restrictToDeclarations ? _invokableAncestorKinds.Add(SyntaxKind.Block).Add(SyntaxKind.ArrowExpressionClause) : _invokableAncestorKinds;
```

Comment:
could also be `.AddRange(new[] {SyntaxKind.Block, SyntaxKind.ArrowExpressionClause})`